src/averageAges.js

```diff
@@ -36,8 +49,16 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  const allWomens = people.filter(human => human.sex === 'f');
+
+  const AgesWomens = withChildren === true
+    ? allWomens.filter(human => people
+      .find(child => child.mother === human.name))
+    : allWomens;
```

Comment:
- You don't need to check whether some boolean var === true. You may use shortcut here
- think of better name for this variable. It should describe what is stored inside and should not be capitalizied.
```suggestion
  const AgesWomens = withChildren
    ? allWomens.filter(human => people
      .find(child => child.mother === human.name))
    : allWomens;
```